syntax = "proto2";

message FUZZING_BYTES {
  required PRIMTIVE_POOL_INIT_VALUES pool_init_values = 1;
  repeated APIB_CALL api_block_call_seq = 2;
}

message APIB_CALL {
  required uint32 api_block_call_num = 1;
  required APIB_CALL_INPUT api_block_call_input = 2;
}

message PRIMTIVE_POOL_INIT_VALUES {
  repeated BUF_UINT8_T_P buf_uint8_t_p = 1;
  repeated _INT _int = 2;
  repeated CURLOPTION curloption = 3;
}

message APIB_CALL_INPUT {
  required uint32 in_idx__size_t_0 = 1;
  required uint32 out_idx__size_t_0 = 2;
  required uint32 in_idx_buf_uint8_t_p_0 = 3;
  required uint32 in_idx__int_0 = 4;
  required uint32 in_idx__int_1 = 5;
  required uint32 out_idx__int_0 = 6;
  required uint32 out_idx__int_1 = 7;
  required uint32 in_idx_FUZZ_DATA_0 = 8;
  required uint32 out_idx_FUZZ_DATA_0 = 9;
  required uint32 in_idx_CURLoption_0 = 10;
  required uint32 in_idx_CURLoption_1 = 11;
  required uint32 in_idx_CURLoption_2 = 12;
  required uint32 in_idx_CURLoption_3 = 13;
  required uint32 out_idx_CURLoption_0 = 14;
  required uint32 out_idx_CURLoption_1 = 15;
  required uint32 out_idx_CURLoption_2 = 16;
  required uint32 out_idx_CURLoption_3 = 17;
  required uint32 in_idx_TLV_0 = 18;
  required uint32 out_idx_TLV_0 = 19;
}

message BUF_UINT8_T_P {
  required bytes val = 1;
}

message _INT {
  required int32 val = 1;
}

enum CURLOPTION {
  val_CURLOPT_WRITEDATA = 1; 
  val_CURLOPT_URL = 2; 
  val_CURLOPT_PORT = 3; 
  val_CURLOPT_PROXY = 4; 
  val_CURLOPT_USERPWD = 5; 
  val_CURLOPT_PROXYUSERPWD = 6; 
  val_CURLOPT_RANGE = 7; 
  val_CURLOPT_READDATA = 8; 
  val_CURLOPT_ERRORBUFFER = 9; 
  val_CURLOPT_WRITEFUNCTION = 10; 
  val_CURLOPT_READFUNCTION = 11; 
  val_CURLOPT_TIMEOUT = 12; 
  val_CURLOPT_INFILESIZE = 13; 
  val_CURLOPT_POSTFIELDS = 14; 
  val_CURLOPT_REFERER = 15; 
  val_CURLOPT_FTPPORT = 16; 
  val_CURLOPT_USERAGENT = 17; 
  val_CURLOPT_LOW_SPEED_LIMIT = 18; 
  val_CURLOPT_LOW_SPEED_TIME = 19; 
  val_CURLOPT_RESUME_FROM = 20; 
  val_CURLOPT_COOKIE = 21; 
  val_CURLOPT_HTTPHEADER = 22; 
  val_CURLOPT_HTTPPOST = 23; 
  val_CURLOPT_SSLCERT = 24; 
  val_CURLOPT_KEYPASSWD = 25; 
  val_CURLOPT_CRLF = 26; 
  val_CURLOPT_QUOTE = 27; 
  val_CURLOPT_HEADERDATA = 28; 
  val_CURLOPT_COOKIEFILE = 29; 
  val_CURLOPT_SSLVERSION = 30; 
  val_CURLOPT_TIMECONDITION = 31; 
  val_CURLOPT_TIMEVALUE = 32; 
  val_CURLOPT_CUSTOMREQUEST = 33; 
  val_CURLOPT_STDERR = 34; 
  val_CURLOPT_POSTQUOTE = 35; 
  val_CURLOPT_OBSOLETE40 = 36; 
  val_CURLOPT_VERBOSE = 37; 
  val_CURLOPT_HEADER = 38; 
  val_CURLOPT_NOPROGRESS = 39; 
  val_CURLOPT_NOBODY = 40; 
  val_CURLOPT_FAILONERROR = 41; 
  val_CURLOPT_UPLOAD = 42; 
  val_CURLOPT_POST = 43; 
  val_CURLOPT_DIRLISTONLY = 44; 
  val_CURLOPT_APPEND = 45; 
  val_CURLOPT_NETRC = 46; 
  val_CURLOPT_FOLLOWLOCATION = 47; 
  val_CURLOPT_TRANSFERTEXT = 48; 
  val_CURLOPT_PUT = 49; 
  val_CURLOPT_PROGRESSFUNCTION = 50; 
  val_CURLOPT_XFERINFODATA = 51; 
  val_CURLOPT_AUTOREFERER = 52; 
  val_CURLOPT_PROXYPORT = 53; 
  val_CURLOPT_POSTFIELDSIZE = 54; 
  val_CURLOPT_HTTPPROXYTUNNEL = 55; 
  val_CURLOPT_INTERFACE = 56; 
  val_CURLOPT_KRBLEVEL = 57; 
  val_CURLOPT_SSL_VERIFYPEER = 58; 
  val_CURLOPT_CAINFO = 59; 
  val_CURLOPT_MAXREDIRS = 60; 
  val_CURLOPT_FILETIME = 61; 
  val_CURLOPT_TELNETOPTIONS = 62; 
  val_CURLOPT_MAXCONNECTS = 63; 
  val_CURLOPT_OBSOLETE72 = 64; 
  val_CURLOPT_FRESH_CONNECT = 65; 
  val_CURLOPT_FORBID_REUSE = 66; 
  val_CURLOPT_RANDOM_FILE = 67; 
  val_CURLOPT_EGDSOCKET = 68; 
  val_CURLOPT_CONNECTTIMEOUT = 69; 
  val_CURLOPT_HEADERFUNCTION = 70; 
  val_CURLOPT_HTTPGET = 71; 
  val_CURLOPT_SSL_VERIFYHOST = 72; 
  val_CURLOPT_COOKIEJAR = 73; 
  val_CURLOPT_SSL_CIPHER_LIST = 74; 
  val_CURLOPT_HTTP_VERSION = 75; 
  val_CURLOPT_FTP_USE_EPSV = 76; 
  val_CURLOPT_SSLCERTTYPE = 77; 
  val_CURLOPT_SSLKEY = 78; 
  val_CURLOPT_SSLKEYTYPE = 79; 
  val_CURLOPT_SSLENGINE = 80; 
  val_CURLOPT_SSLENGINE_DEFAULT = 81; 
  val_CURLOPT_DNS_USE_GLOBAL_CACHE = 82; 
  val_CURLOPT_DNS_CACHE_TIMEOUT = 83; 
  val_CURLOPT_PREQUOTE = 84; 
  val_CURLOPT_DEBUGFUNCTION = 85; 
  val_CURLOPT_DEBUGDATA = 86; 
  val_CURLOPT_COOKIESESSION = 87; 
  val_CURLOPT_CAPATH = 88; 
  val_CURLOPT_BUFFERSIZE = 89; 
  val_CURLOPT_NOSIGNAL = 90; 
  val_CURLOPT_SHARE = 91; 
  val_CURLOPT_PROXYTYPE = 92; 
  val_CURLOPT_PRIVATE = 93; 
  val_CURLOPT_HTTP200ALIASES = 94; 
  val_CURLOPT_UNRESTRICTED_AUTH = 95; 
  val_CURLOPT_FTP_USE_EPRT = 96; 
  val_CURLOPT_HTTPAUTH = 97; 
  val_CURLOPT_SSL_CTX_FUNCTION = 98; 
  val_CURLOPT_SSL_CTX_DATA = 99; 
  val_CURLOPT_FTP_CREATE_MISSING_DIRS = 100; 
  val_CURLOPT_PROXYAUTH = 101; 
  val_CURLOPT_SERVER_RESPONSE_TIMEOUT = 102; 
  val_CURLOPT_IPRESOLVE = 103; 
  val_CURLOPT_MAXFILESIZE = 104; 
  val_CURLOPT_INFILESIZE_LARGE = 105; 
  val_CURLOPT_RESUME_FROM_LARGE = 106; 
  val_CURLOPT_MAXFILESIZE_LARGE = 107; 
  val_CURLOPT_NETRC_FILE = 108; 
  val_CURLOPT_USE_SSL = 109; 
  val_CURLOPT_POSTFIELDSIZE_LARGE = 110; 
  val_CURLOPT_TCP_NODELAY = 111; 
  val_CURLOPT_FTPSSLAUTH = 112; 
  val_CURLOPT_IOCTLFUNCTION = 113; 
  val_CURLOPT_FTP_ACCOUNT = 114; 
  val_CURLOPT_COOKIELIST = 115; 
  val_CURLOPT_IGNORE_CONTENT_LENGTH = 116; 
  val_CURLOPT_FTP_SKIP_PASV_IP = 117; 
  val_CURLOPT_FTP_FILEMETHOD = 118; 
  val_CURLOPT_LOCALPORT = 119; 
  val_CURLOPT_LOCALPORTRANGE = 120; 
  val_CURLOPT_CONNECT_ONLY = 121; 
  val_CURLOPT_CONV_FROM_NETWORK_FUNCTION = 122; 
  val_CURLOPT_CONV_TO_NETWORK_FUNCTION = 123; 
  val_CURLOPT_CONV_FROM_UTF8_FUNCTION = 124; 
  val_CURLOPT_MAX_SEND_SPEED_LARGE = 125; 
  val_CURLOPT_MAX_RECV_SPEED_LARGE = 126; 
  val_CURLOPT_FTP_ALTERNATIVE_TO_USER = 127; 
  val_CURLOPT_SOCKOPTFUNCTION = 128; 
  val_CURLOPT_SOCKOPTDATA = 129; 
  val_CURLOPT_SSL_SESSIONID_CACHE = 130; 
  val_CURLOPT_SSH_AUTH_TYPES = 131; 
  val_CURLOPT_SSH_PUBLIC_KEYFILE = 132; 
  val_CURLOPT_SSH_PRIVATE_KEYFILE = 133; 
  val_CURLOPT_FTP_SSL_CCC = 134; 
  val_CURLOPT_TIMEOUT_MS = 135; 
  val_CURLOPT_CONNECTTIMEOUT_MS = 136; 
  val_CURLOPT_HTTP_TRANSFER_DECODING = 137; 
  val_CURLOPT_HTTP_CONTENT_DECODING = 138; 
  val_CURLOPT_NEW_FILE_PERMS = 139; 
  val_CURLOPT_NEW_DIRECTORY_PERMS = 140; 
  val_CURLOPT_SSH_HOST_PUBLIC_KEY_MD5 = 141; 
  val_CURLOPT_OPENSOCKETFUNCTION = 142; 
  val_CURLOPT_OPENSOCKETDATA = 143; 
  val_CURLOPT_COPYPOSTFIELDS = 144; 
  val_CURLOPT_PROXY_TRANSFER_MODE = 145; 
  val_CURLOPT_SEEKFUNCTION = 146; 
  val_CURLOPT_SEEKDATA = 147; 
  val_CURLOPT_CRLFILE = 148; 
  val_CURLOPT_ISSUERCERT = 149; 
  val_CURLOPT_ADDRESS_SCOPE = 150; 
  val_CURLOPT_CERTINFO = 151; 
  val_CURLOPT_USERNAME = 152; 
  val_CURLOPT_PASSWORD = 153; 
  val_CURLOPT_PROXYUSERNAME = 154; 
  val_CURLOPT_PROXYPASSWORD = 155; 
  val_CURLOPT_NOPROXY = 156; 
  val_CURLOPT_TFTP_BLKSIZE = 157; 
  val_CURLOPT_SOCKS5_GSSAPI_SERVICE = 158; 
  val_CURLOPT_SOCKS5_GSSAPI_NEC = 159; 
  val_CURLOPT_PROTOCOLS = 160; 
  val_CURLOPT_REDIR_PROTOCOLS = 161; 
  val_CURLOPT_SSH_KNOWNHOSTS = 162; 
  val_CURLOPT_SSH_KEYFUNCTION = 163; 
  val_CURLOPT_SSH_KEYDATA = 164; 
  val_CURLOPT_MAIL_FROM = 165; 
  val_CURLOPT_MAIL_RCPT = 166; 
  val_CURLOPT_FTP_USE_PRET = 167; 
  val_CURLOPT_RTSP_REQUEST = 168; 
  val_CURLOPT_RTSP_SESSION_ID = 169; 
  val_CURLOPT_RTSP_STREAM_URI = 170; 
  val_CURLOPT_RTSP_TRANSPORT = 171; 
  val_CURLOPT_RTSP_CLIENT_CSEQ = 172; 
  val_CURLOPT_RTSP_SERVER_CSEQ = 173; 
  val_CURLOPT_INTERLEAVEDATA = 174; 
  val_CURLOPT_INTERLEAVEFUNCTION = 175; 
  val_CURLOPT_WILDCARDMATCH = 176; 
  val_CURLOPT_CHUNK_BGN_FUNCTION = 177; 
  val_CURLOPT_CHUNK_END_FUNCTION = 178; 
  val_CURLOPT_FNMATCH_FUNCTION = 179; 
  val_CURLOPT_CHUNK_DATA = 180; 
  val_CURLOPT_FNMATCH_DATA = 181; 
  val_CURLOPT_RESOLVE = 182; 
  val_CURLOPT_TLSAUTH_USERNAME = 183; 
  val_CURLOPT_TLSAUTH_PASSWORD = 184; 
  val_CURLOPT_TLSAUTH_TYPE = 185; 
  val_CURLOPT_TRANSFER_ENCODING = 186; 
  val_CURLOPT_CLOSESOCKETFUNCTION = 187; 
  val_CURLOPT_CLOSESOCKETDATA = 188; 
  val_CURLOPT_GSSAPI_DELEGATION = 189; 
  val_CURLOPT_DNS_SERVERS = 190; 
  val_CURLOPT_ACCEPTTIMEOUT_MS = 191; 
  val_CURLOPT_TCP_KEEPALIVE = 192; 
  val_CURLOPT_TCP_KEEPIDLE = 193; 
  val_CURLOPT_TCP_KEEPINTVL = 194; 
  val_CURLOPT_SSL_OPTIONS = 195; 
  val_CURLOPT_MAIL_AUTH = 196; 
  val_CURLOPT_SASL_IR = 197; 
  val_CURLOPT_XFERINFOFUNCTION = 198; 
  val_CURLOPT_XOAUTH2_BEARER = 199; 
  val_CURLOPT_DNS_INTERFACE = 200; 
  val_CURLOPT_DNS_LOCAL_IP4 = 201; 
  val_CURLOPT_DNS_LOCAL_IP6 = 202; 
  val_CURLOPT_LOGIN_OPTIONS = 203; 
  val_CURLOPT_SSL_ENABLE_NPN = 204; 
  val_CURLOPT_SSL_ENABLE_ALPN = 205; 
  val_CURLOPT_EXPECT_100_TIMEOUT_MS = 206; 
  val_CURLOPT_PROXYHEADER = 207; 
  val_CURLOPT_HEADEROPT = 208; 
  val_CURLOPT_PINNEDPUBLICKEY = 209; 
  val_CURLOPT_UNIX_SOCKET_PATH = 210; 
  val_CURLOPT_SSL_VERIFYSTATUS = 211; 
  val_CURLOPT_SSL_FALSESTART = 212; 
  val_CURLOPT_PATH_AS_IS = 213; 
  val_CURLOPT_PROXY_SERVICE_NAME = 214; 
  val_CURLOPT_SERVICE_NAME = 215; 
  val_CURLOPT_PIPEWAIT = 216; 
  val_CURLOPT_DEFAULT_PROTOCOL = 217; 
  val_CURLOPT_STREAM_WEIGHT = 218; 
  val_CURLOPT_STREAM_DEPENDS = 219; 
  val_CURLOPT_STREAM_DEPENDS_E = 220; 
  val_CURLOPT_TFTP_NO_OPTIONS = 221; 
  val_CURLOPT_CONNECT_TO = 222; 
  val_CURLOPT_TCP_FASTOPEN = 223; 
  val_CURLOPT_KEEP_SENDING_ON_ERROR = 224; 
  val_CURLOPT_PROXY_CAINFO = 225; 
  val_CURLOPT_PROXY_CAPATH = 226; 
  val_CURLOPT_PROXY_SSL_VERIFYPEER = 227; 
  val_CURLOPT_PROXY_SSL_VERIFYHOST = 228; 
  val_CURLOPT_PROXY_SSLVERSION = 229; 
  val_CURLOPT_PROXY_TLSAUTH_USERNAME = 230; 
  val_CURLOPT_PROXY_TLSAUTH_PASSWORD = 231; 
  val_CURLOPT_PROXY_TLSAUTH_TYPE = 232; 
  val_CURLOPT_PROXY_SSLCERT = 233; 
  val_CURLOPT_PROXY_SSLCERTTYPE = 234; 
  val_CURLOPT_PROXY_SSLKEY = 235; 
  val_CURLOPT_PROXY_SSLKEYTYPE = 236; 
  val_CURLOPT_PROXY_KEYPASSWD = 237; 
  val_CURLOPT_PROXY_SSL_CIPHER_LIST = 238; 
  val_CURLOPT_PROXY_CRLFILE = 239; 
  val_CURLOPT_PROXY_SSL_OPTIONS = 240; 
  val_CURLOPT_PRE_PROXY = 241; 
  val_CURLOPT_PROXY_PINNEDPUBLICKEY = 242; 
  val_CURLOPT_ABSTRACT_UNIX_SOCKET = 243; 
  val_CURLOPT_SUPPRESS_CONNECT_HEADERS = 244; 
  val_CURLOPT_REQUEST_TARGET = 245; 
  val_CURLOPT_SOCKS5_AUTH = 246; 
  val_CURLOPT_SSH_COMPRESSION = 247; 
  val_CURLOPT_MIMEPOST = 248; 
  val_CURLOPT_TIMEVALUE_LARGE = 249; 
  val_CURLOPT_HAPPY_EYEBALLS_TIMEOUT_MS = 250; 
  val_CURLOPT_RESOLVER_START_FUNCTION = 251; 
  val_CURLOPT_RESOLVER_START_DATA = 252; 
  val_CURLOPT_HAPROXYPROTOCOL = 253; 
  val_CURLOPT_DNS_SHUFFLE_ADDRESSES = 254; 
  val_CURLOPT_TLS13_CIPHERS = 255; 
  val_CURLOPT_PROXY_TLS13_CIPHERS = 256; 
  val_CURLOPT_DISALLOW_USERNAME_IN_URL = 257; 
  val_CURLOPT_DOH_URL = 258; 
  val_CURLOPT_UPLOAD_BUFFERSIZE = 259; 
  val_CURLOPT_UPKEEP_INTERVAL_MS = 260; 
  val_CURLOPT_CURLU = 261; 
  val_CURLOPT_TRAILERFUNCTION = 262; 
  val_CURLOPT_TRAILERDATA = 263; 
  val_CURLOPT_HTTP09_ALLOWED = 264; 
  val_CURLOPT_ALTSVC_CTRL = 265; 
  val_CURLOPT_ALTSVC = 266; 
  val_CURLOPT_MAXAGE_CONN = 267; 
  val_CURLOPT_SASL_AUTHZID = 268; 
  val_CURLOPT_MAIL_RCPT_ALLOWFAILS = 269; 
  val_CURLOPT_SSLCERT_BLOB = 270; 
  val_CURLOPT_SSLKEY_BLOB = 271; 
  val_CURLOPT_PROXY_SSLCERT_BLOB = 272; 
  val_CURLOPT_PROXY_SSLKEY_BLOB = 273; 
  val_CURLOPT_ISSUERCERT_BLOB = 274; 
  val_CURLOPT_PROXY_ISSUERCERT = 275; 
  val_CURLOPT_PROXY_ISSUERCERT_BLOB = 276; 
  val_CURLOPT_SSL_EC_CURVES = 277; 
  val_CURLOPT_HSTS_CTRL = 278; 
  val_CURLOPT_HSTS = 279; 
  val_CURLOPT_HSTSREADFUNCTION = 280; 
  val_CURLOPT_HSTSREADDATA = 281; 
  val_CURLOPT_HSTSWRITEFUNCTION = 282; 
  val_CURLOPT_HSTSWRITEDATA = 283; 
  val_CURLOPT_AWS_SIGV4 = 284; 
  val_CURLOPT_DOH_SSL_VERIFYPEER = 285; 
  val_CURLOPT_DOH_SSL_VERIFYHOST = 286; 
  val_CURLOPT_DOH_SSL_VERIFYSTATUS = 287; 
  val_CURLOPT_CAINFO_BLOB = 288; 
  val_CURLOPT_PROXY_CAINFO_BLOB = 289; 
  val_CURLOPT_SSH_HOST_PUBLIC_KEY_SHA256 = 290; 
  val_CURLOPT_PREREQFUNCTION = 291; 
  val_CURLOPT_PREREQDATA = 292; 
  val_CURLOPT_MAXLIFETIME_CONN = 293; 
  val_CURLOPT_MIME_OPTIONS = 294; 
  val_CURLOPT_SSH_HOSTKEYFUNCTION = 295; 
  val_CURLOPT_SSH_HOSTKEYDATA = 296; 
  val_CURLOPT_PROTOCOLS_STR = 297; 
  val_CURLOPT_REDIR_PROTOCOLS_STR = 298; 
  val_CURLOPT_WS_OPTIONS = 299; 
  val_CURLOPT_CA_CACHE_TIMEOUT = 300; 
  val_CURLOPT_QUICK_EXIT = 301; 
  val_CURLOPT_SERVER_RESPONSE_TIMEOUT_MS = 302; 
  val_CURLOPT_ECH = 303; 
  val_CURLOPT_TCP_KEEPCNT = 304; 
  val_CURLOPT_LASTENTRY = 305; 
}
